rfcs/20180905-deprecate-collections.md

```diff
@@ -0,0 +1,575 @@
+  <tr>
+   <td>Summaries
+   </td>
+   <td>_SUMMARY_WRITER_INIT_COLLECTION_NAME
+   </td>
+   <td>Global variable
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Summaries
+   </td>
+   <td>SUMMARIES
+   </td>
+   <td>Not needed with SummaryV2
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Summaries
+   </td>
+   <td>SUMMARY_OP
+   </td>
+   <td>Not needed with SummaryV2
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Queues
+   </td>
+   <td>QUEUE_RUNNERS
+   </td>
+   <td>Not needed with tf.data, Distribution strategies
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Losses
+   </td>
+   <td>REGULARIZATION_LOSSES
+   </td>
+   <td>Not needed after variables 2.0
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Losses
+   </td>
+   <td>LOSSES
+   </td>
+   <td>Model.losses
+   </td>
+   <td>Use Losses returned from functions in loss_impl.py
+   </td>
+  </tr>
+  <tr>
+   <td>Saving
+   </td>
+   <td>SAVEABLE_OBJECTS
+   </td>
+   <td>Not needed with object based checkpointing
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Saving
+   </td>
+   <td>SAVERS
+   </td>
+   <td>Not needed with object based checkpointing
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Conditionals
+   </td>
+   <td>COND_CONTEXT, WHILE_CONTEXT
+   </td>
+   <td>Not needed with CondV2
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Assets
+   </td>
+   <td>ASSET_FILEPATHS
+   </td>
+   <td>Keras backend
+   </td>
+   <td>table_creator_scope
+   </td>
+  </tr>
+  <tr>
+   <td>Resources
+   </td>
+   <td>RESOURCES, LOCAL_RESOURCES
+   </td>
+   <td>Not used much
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>Updates
+   </td>
+   <td>UPDATE_OPS
+   </td>
+   <td>Model.updates
+   </td>
+   <td>BatchNormalization.updates
+   </td>
+  </tr>
+  <tr>
+   <td>Iterators
+   </td>
+   <td>GLOBAL_ITERATORS
+   </td>
+   <td>Object based checkpointing handles saveables generically
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>SavedModel
+   </td>
+   <td>INIT_OP
+   </td>
+   <td>Keep
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>SavedModel
+   </td>
+   <td>LOCAL_INIT_OP
+   </td>
+   <td>Keep
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>SavedModel
+   </td>
+   <td>READY_OP
+   </td>
+   <td>Keep
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>SavedModel
+   </td>
+   <td>READY_FOR_LOCAL_INIT_OP
+   </td>
+   <td>Keep
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>SavedModel
+   </td>
+   <td>GLOBAL_STEP
+   </td>
+   <td>Keep
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+For some specific use cases, we'll elaborate how we will handle them
+
+
+### **Global variables, trainable variables**
+
+For the keras models use case, Model.variables already tracks all variables created in a Keras model. Similarly, for trainable variables, Model.trainable_variables tracks all trainable variables. 
+
+In case of custom graph code, variable_creator_scope can be used to collect up variables.
```

Comment:
Added some example code. Regarding estimators: I think we can create one of these variable trackers before the model_fn is called, record the variables created and then feed them to the Scaffold so that the initialization happens correctly and the init_op etc. is set correctly.